test/widgets/new_dm_sheet_test.dart

```diff
@@ -0,0 +1,293 @@
+        check(find.text(eg.selfUser.fullName)).findsNothing();
+      });
+
+      testWidgets('other user selection hides self user', (tester) async {
+        final otherUser = eg.user(fullName: 'Other User');
+        final otherUserTileFinder = find.ancestor(
+          of: find.text(otherUser.fullName),
+          matching: find.byType(InkWell));
+        await setupSheet(tester, users: [otherUser]);
+
+        check(find.text(eg.selfUser.fullName)).findsOne();
+
+        await tester.tap(otherUserTileFinder);
+        await tester.pump();
+        check(find.text(eg.selfUser.fullName)).findsNothing();
+      });
+
+      testWidgets('can select multiple users', (tester) async {
+        final user1 = eg.user(fullName: 'Test User 1');
+        final user2 = eg.user(fullName: 'Test User 2');
+        await setupSheet(tester, users: [user1, user2]);
+
+        final userTile1 = find.ancestor(
+          of: find.text(user1.fullName),
+          matching: find.byType(InkWell));
+        final userTile2 = find.ancestor(
+          of: find.text(user2.fullName),
+          matching: find.byType(InkWell));
+
+        await tester.tap(userTile1);
+        await tester.pump();
+        await tester.tap(userTile2);
+        await tester.pump();
+        check(find.byIcon(Icons.check_circle)).findsExactly(2);
+      });
+    });
+
+    group('navigation to DM Narrow', () {
+      Future<void> runAndCheck(WidgetTester tester, {
+        required List<User> users,
+        required String expectedAppBarTitle,
+        bool isSelfDm = false,
+      }) async {
+        await setupSheet(tester, users: users);
+
+        final context = tester.element(find.byType(NewDmPicker));
+        final store = PerAccountStoreWidget.of(context);
+        final connection = store.connection as FakeApiConnection;
+
+        connection.prepare(
+          json: eg.newestGetMessagesResult(foundOldest: true, messages: []).toJson());
+        for (final user in users) {
+          final userTile = find.ancestor(
+            of: find.text(user.fullName),
+            matching: find.byType(InkWell));
+          await tester.tap(userTile);
+          await tester.pump();
+        }
+        if (isSelfDm) {
+          final userTile = find.ancestor(
+            of: find.text(eg.selfUser.fullName),
+            matching: find.byType(InkWell));
+          await tester.tap(userTile);
+          await tester.pump();
+        }
+        await tester.tap(find.widgetWithText(GestureDetector, 'Next'));
+        await tester.pumpAndSettle();
+        check(find.descendant(
+          of: find.byType(ZulipAppBar),
+          matching: find.text(expectedAppBarTitle))).findsOne();
+
+        check(find.byType(ComposeBox)).findsOne();
+      }
+
+      testWidgets('navigates to self DM on Next', (tester) async {
+        await runAndCheck(
+          tester,
+          users: [],
+          expectedAppBarTitle: 'DMs with yourself',
+          isSelfDm: true);
```

Comment:
Hmm, how about just passing `users: [eg.selfUser]`? That way we don't need to introduce a separate parameter.